gapis/service/service.proto

```diff
@@ -1287,6 +1290,15 @@ message TraceTypeCapabilities {
   bool requires_application = 6;
 }
 
+message GpuProfileRequest {
+  path.Capture capture = 1;
+  path.Device device = 2;
+  TraceOptions trace_options = 3;
```

Comment:
Don't pass the trace options here. That is way too generic. I think the only thing we need here are the capture and the device.

Another reason why passing the perfetto config from the client doesn't work here: you'd have to setup all the layers (not just the GPU slices layer) when starting gapir.

I think all you have to do is move the code that creates the trace options from gapit into gapis.

Reply:
We shouldn't have to do much for the layers, we could put "all" of the layers in the right place when starting GAPIR always, then we can turn them on/off at instance creation time I think.

Reply:
The system setting overrides the `VkInstanceCreateInfo`. I don't think you can "turn-off" any layers the system requests.

Reply:
But we can **not** add these layers to the system setting, and just add them to the VkInstanceCreateInfo (in a transform).